hlx_statics/blocks/buttonblock/buttonblock.js

```diff
@@ -0,0 +1,31 @@
+import { decorateButtons } from "../../scripts/lib-adobeio.js";
+
+/**
+ * Decorates the button block.
+ * @param {Element} block The calendar block element.
+ */
+export default async function decorate(block) {

```

Comment:
I'm not sure if we should create a `buttonblock`. Looking at the markdown file for the test url (https://button-block--adp-devsite--adobedocs.aem.page/developer-distribution/creative-cloud/docs/guides/zxp/distribution), it looks like the button is part of `TeaserBlock`: https://github.com/AdobeDocs/developer-distribute/blob/main/src/pages/guides/zxp/distribution.md?plain=1#L31-L33

I think this logic should reside within  `decorateButtons` instead. I believe @louisachu had to do something similar for `mini-resource-card` buttons (https://github.com/AdobeDocs/adp-devsite/pull/39). @louisachu what are your thoughts?

Reply:
@BaskarMitrah after talking with @louisachu, I think I understand why you created `ButtonBlock`. I see in the markdown file, you've replaced [TeaserBlock](https://github.com/AdobeDocs/developer-distribute/blob/main/src/pages/guides/zxp/distribution.md?plain=1#L31) with [ButtonBlock](https://github.com/AdobeDocs/developer-distribute/blob/eds-migration/src/pages/guides/zxp/distribution.md?plain=1#L30).

TeaserBlock was actually merged into Announcement Block (https://jira.corp.adobe.com/browse/DEVSITE-1428).

So could you try
- replacing ButtonBlock with Announcement block here:  https://github.com/AdobeDocs/developer-distribute/blob/eds-migration/src/pages/guides/zxp/distribution.md?plain=1#L30
- move ButtonBlock logic within decorateButtons, if still needed
- delete ButtonBlock
- and see if that achieves the desired result?

Reply:
OKay @melissag-ensemble